adr/ADR-19.md

```diff
@@ -0,0 +1,281 @@
+- Multiple objects can be placed in each bucket
+- Object Info is stored as json in the payload of the message on the Object Info message subject. 
+- The Object Info subject is always rolled up (per subject)
+- Object chunks are stored as the payload of messages on the Chunk message subject
+
+## Structures
+
+### ObjectStoreConfig
+
+The object store config is the basis for the stream configuration.
+```go
+type ObjectStoreConfig struct {
+    Bucket      string
+    Description string
+    TTL         time.Duration
+    Storage     StorageType
+    Replicas    int
+}
+```
+
+### ObjectLink is used to embed links to other buckets and objects.
+
+```go
+type ObjectLink struct {
+    // Bucket is the name of the other object store.
+    Bucket string `json:"bucket"`
+    
+    // Name can be used to link to a single object.
+    // If empty means this is a link to the whole store, like a directory.
+    Name string `json:"name,omitempty"`
+}
+```
+
+### ObjectMetaOptions
+
+```go
+type ObjectMetaOptions struct {
+    Link      *ObjectLink `json:"link,omitempty"`
+    ChunkSize uint32      `json:"max_chunk_size,omitempty"`
+}
+```
+
+### ObjectMeta 
+
+Object Meta is high level information about an object.
+
+```go
+type ObjectMeta struct {
+    Name        string `json:"name"`
+    Description string `json:"description,omitempty"`
+    Headers     Header `json:"headers,omitempty"`
+
+    // Optional options.
+    Opts *ObjectMetaOptions `json:"options,omitempty"`
+}
+```
+
+### ObjectInfo 
+
+Object Info is meta plus instance information.
+
+```go
+type ObjectInfo struct {
+    ObjectMeta
+    
+    Bucket  string    `json:"bucket"`
+    
+    NUID    string    `json:"nuid"`
+    
+    // the total object size in bytes
+    Size    uint64    `json:"size"`
+    
+    ModTime time.Time `json:"mtime"`
+    
+    // the total number of chunks
+    Chunks  uint32    `json:"chunks"`
+    
+    // as in http, <digest-algorithm>=<digest-value>
+    Digest  string    `json:"digest,omitempty"`
+    
+    Deleted bool      `json:"deleted,omitempty"`
+}
+```
+
+### ObjectStoreStatus
+
+The status of an object
+
+```go
+type ObjectStoreStatus interface {
+    // Bucket is the name of the bucket
+    Bucket() string
+    
+    // Description is the description supplied when creating the bucket
+    Description() string
+    
+    // TTL indicates how long objects are kept in the bucket
+    TTL() time.Duration
+    
+    // Storage indicates the underlying JetStream storage technology used to store data
+    Storage() StorageType
+    
+    // Replicas indicates how many storage replicas are kept for the data in the bucket
+    Replicas() int
+    
+    // Sealed indicates the stream is sealed and cannot be modified in any way
+    Sealed() bool
+    
+    // Size is the combined size of all data in the bucket including metadata, in bytes
+    Size() uint64
+    
+    // BackingStore provides details about the underlying storage
```

Comment:
For now the only valid value is `JetStream` later we might add or other backing stores.

Reply:
will add a commment